xbmc/windowing/X11/WinSystemX11.cpp

```diff
@@ -1070,4 +1072,51 @@ void CWinSystemX11::UpdateCrtc()
   g_graphicsContext.SetFPS(fps);
 }
 
+std::string CWinSystemX11::GetClipboardText(void)
+{
+  if (!m_dpy)
+     return "";
+
+  std::string result;
+  Atom utf8_string = XInternAtom(m_dpy, "UTF8_STRING", False);
+  Atom clipboard = XInternAtom(m_dpy, "CLIPBOARD", False);
+  XConvertSelection(m_dpy, clipboard, utf8_string, None, m_glWindow, CurrentTime);
+//  Wait for clipboard event (SelectionNotify) sync, code copied from
+//  https://github.com/vinniefalco/SimpleDJ/blob/master/Extern/JUCE/modules/juce_gui_basics/native/juce_linux_Clipboard.cpp
+//  clipboard content requesting is inherently slow on x11, it
+//  often takes 50ms or more so...
+  int count = 20; // will wait at most for 200 ms
+  while (--count >= 0)
+  {
+    if (CWinEventsX11Imp::MessagePump())
+      break;
+    Sleep(10);
+  }
+
+  for(unsigned long offset = 0;;)
```

Comment:
oh :) this is really a funny style of a loop
if you did:
while (itemsLeft)
it would be much clearer

and the used parameters should be outside of the loop